src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientAnalyzerBase.cs

```diff
@@ -26,16 +26,26 @@ public override void Analyze(ISymbolAnalysisContext context)
             AnalyzeCore(context);
         }
 
+        protected class ParameterEquivalenceComparerOptionalIgnore : ParameterEquivalenceComparer
+        {
+            public static new ParameterEquivalenceComparerOptionalIgnore Default { get; } = new ParameterEquivalenceComparerOptionalIgnore();
+
+            public override bool Equals(IParameterSymbol x, IParameterSymbol y)
+            {
+                return x.Name.Equals(y.Name) && TypeSymbolEquals(x.Type, y.Type);
+            }
+        }
+
         protected class ParameterEquivalenceComparer : IEqualityComparer<IParameterSymbol>, IEqualityComparer<ITypeParameterSymbol>
         {
             public static ParameterEquivalenceComparer Default { get; } = new ParameterEquivalenceComparer();
 
-            public bool Equals(IParameterSymbol x, IParameterSymbol y)
+            public virtual bool Equals(IParameterSymbol x, IParameterSymbol y)
             {
                 return x.Name.Equals(y.Name) && TypeSymbolEquals(x.Type, y.Type) && x.IsOptional == y.IsOptional;
             }
 
-            private bool TypeSymbolEquals(ITypeSymbol x, ITypeSymbol y)
+            protected bool TypeSymbolEquals(ITypeSymbol x, ITypeSymbol y)
```

Comment:
We should consider making this static or virtual, depending on whether we see a need to override in the future.

Reply:
I would prefer not to change it until we see a real need.

Reply:
If not virtual, there's no benefit to this being an instance member and no risk to elevating it.    Unless we have a reason to keep it from being overridden, let's make it virtual, please.

Reply:
Makes sense. Updated.